src/pages/iou/request/step/IOURequestStepMerchant.tsx

```diff
@@ -83,7 +87,12 @@ function IOURequestStepMerchant({
         [isMerchantRequired, translate],
     );
 
+    const updateMerchantRef = (value: string) => {
+        merchantRef.current = value;
+    };
+
```

Comment:
I used ref here (same for desc) so it won't trigger a re-render everytime the value is updated since we only use this for `getHasUnsavedChanges`.